cmd/poller/collector/collector.go

```diff
@@ -235,7 +236,11 @@ func Init(c Collector) error {
 
 // Start will run the collector in an infinity loop
 func (me *AbstractCollector) Start(wg *sync.WaitGroup) {
-
+	defer func() {
+		if r := recover(); r != nil {
+			me.Logger.Error().Stack().Err(errors.New(errors.GO_ROUTINE_PANIC, string(debug.Stack()))).Msg("Collector panicked")
```

Comment:
Don't we already have stacks with [zerolog](https://github.com/rs/zerolog#error-logging-with-stacktrace)?

Reply:
recover return type is not error. One way to get stacktrace is like this.